### DISCUSSION TITLE
Popup location and size

### DISCUSSION THREAD

**Discussion by pfw** [unpoly/unpoly#279] [General]
I'm looking to create a popup that works something like this:

<img width="479" alt="Screen Shot 2021-09-03 at 5 23 39 pm" src="https://user-images.githubusercontent.com/92163/131966908-b54590a8-f988-4f19-bc19-a7d76b2de755.png">

The constraints I'm after are that it appears below (but could ideally adjust exact position) the triggering element and is sized so that the width matches the triggering element.

I'm having a look at the 'Popup position' docs and I'm not sure that they will do what I need with the width - has anyone done something like this before?

**Comment by adam12** [comment]
The challenge will be that the `up-popup` element is rendered at the bottom of the `<body>` and then affixed absolutely, so it has no idea of a parent width (ie. if you rendered `<div style="width: 10rem"><input style="width: 100%">` and then the popup was inserted after the `<input>` in the DOM).

You can achieve this by manually adjusting the width of the `<up-popup>` element to match the width of the `<input>`, by using one of the callbacks. Here's a quick [Glitch](https://glitch.com/edit/#!/amber-patch-bath) I made to demo that.

**Reply by pfw** [reply]
Thanks Adam - I was thinking that it would need something like that, it will certainly fit in as I'll be hooking up some key handlers as well so would have needed some JS anyway.

**Comment by triskweline** [comment]
Unpoly popups may not be a great solution for this use case, where the user is interacting with the parent layer. E.g. you may run into issues with popups capturing focus while the user is trying to type into the parent layer.

I've personally only used Unpoly popups for simple dropdown menus so far.

**Reply by pfw** [reply]
Interesting. For this use case I'm thinking it might be ok since we'd definitely want to capture some specific keys and handle in the new layer but also have any interaction outside of those close the layer automatically. I'll have a poke around and see how it goes.
